sdk/python/tests/unit/api/test_search_api.py

```diff
@@ -0,0 +1,2407 @@
+        assert data["results"] == []
+        assert (
+            data["error"] == "Following projects do not exist: nonexistent_project_xyz"
+        )
+
+    @pytest.mark.parametrize(
+        "parameter_type,test_cases",
+        [
+            (
+                "sorting",
+                [
+                    ("sort_by", "invalid_sort_field", "sort_order", "desc", 400),
+                    ("sort_by", "name", "sort_order", "invalid_order", 400),
+                    ("sort_by", "", "sort_order", "asc", 200),
+                    ("sort_by", "match_score", "sort_order", "", 200),
+                    ("sort_by", "123", "sort_order", "xyz", 400),
+                ],
+            ),
+            (
+                "boolean",
+                [
+                    (
+                        "allow_cache",
+                        "invalid_bool",
+                        None,
+                        None,
+                        422,
+                    ),  # FastAPI may handle gracefully
+                    (
+                        "allow_cache",
+                        "yes",
+                        None,
+                        None,
+                        200,
+                    ),  # FastAPI converts to boolean
+                    (
+                        "allow_cache",
+                        "1",
+                        None,
+                        None,
+                        200,
+                    ),  # FastAPI converts to boolean
+                ],
+            ),
+        ],
+    )
+    def test_search_with_invalid_parameters(
+        self, search_test_app, parameter_type, test_cases
+    ):
+        """Test search API with various invalid parameter combinations"""
+        for param1, value1, param2, value2, expected_code in test_cases:
+            # Build query string
+            query_parts = ["query=user"]
+            query_parts.append(f"{param1}={value1}")
+            if param2 is not None and value2 is not None:
+                query_parts.append(f"{param2}={value2}")
+
+            url = "/search?" + "&".join(query_parts)
+            response = search_test_app.get(url)
+
+            assert response.status_code == expected_code, (
+                f"Expected {expected_code} but got {response.status_code} for {param1}='{value1}'"
+                + (f", {param2}='{value2}'" if param2 else "")
+            )
+
+            if response.status_code == 200:
+                # If successful, verify response format
+                data = response.json()
+                assert "results" in data
+                assert isinstance(data["results"], list)
+            elif response.status_code in [400, 422]:
+                # If validation error, verify it's a proper FastAPI error
+                error_data = response.json()
+                assert "detail" in error_data
+
+    def test_search_with_malicious_query_injection_attempts(self, search_test_app):
```

Comment:
I think we dont need to handle this explicitely as we are not identifying / fixing malcious queries.